lib/haskell/natural4/src/LS/XPile/MathLang/GenericMathLang/GenericMathLangAST.hs

```diff
@@ -184,10 +235,24 @@ exprsToSeqExp :: [Exp] -> SeqExp
 exprsToSeqExp = coerce
 
 consSE :: Exp -> SeqExp -> SeqExp
-consSE expr (seqExpToExprs -> exprs) = exprsToSeqExp $ expr : exprs
-
--- removed GADTs because had been experimenting with `unbound-generics` and didn't know how to get them to work well tgt
--- | May want to put the `md`s back into BaseExp and collapse Exp and BaseExp back into one data structure. Not sure what's more ergo rn
+consSE expr (SeqExp exprs) = exprsToSeqExp $ expr : exprs
+
+-- | Un-annotated GML expression.
+--
+-- The recursive calls are all to 'Exp', which pairs a base expression with metadata.
+-- So we have an "annotated fixed point" construction here, but with the annotation
+-- being fixed to always be metadata.
+--
+-- Notes:
+--
+-- - It is unclear why the operators are separated from each other.
+-- - It is unclear why And / Or have separate constructors instead of being operators.
```

Comment:
Re 

> It is unclear why And / Or have separate constructors instead of being operators.

There probably wasn't a good reason! We / I can make them operators if that's better!

We could also just refactor the AST to be more like the one you had proposed.

Reply:
Ok, thanks. Leaving such possible refactorings for later.